crypto/src/lib.rs

```diff
@@ -148,20 +160,62 @@ pub struct KeyPair {
     pub private_key: PrivateKey,
 }
 
+/// Error when dealing with cryptographic functions
+#[cfg(feature = "std")]
+#[derive(Debug, Display)]
+pub enum Error {
+    /// Returned when trying to create an algorithm which does not exist
+    NoSuchAlgorithm,
+    /// Occurs during deserialization of a private or public key
+    Parse(String),
+    /// Returned when an error occurs during the signing process
+    Signing(String),
+    /// Returned when an error occurs during key generation
+    KeyGen(String),
+    /// Returned when an error occurs during digest generation
+    DigestGen(String),
+    /// A General purpose error message that doesn't fit in any category
+    Other(String),
```

Comment:
Not a huge problem, but I think that even having a general error message encourages people not to use the specific error types. If we insist on having `Other` I suggest having it include an `eyre::Report`, because then at least it can be debugged.